flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/asyncprocessing/AbstractAsyncStateStreamOperator.java

```diff
@@ -74,15 +76,20 @@ public void setup(
         final int asyncBufferSize = environment.getExecutionConfig().getAsyncStateBufferSize();
         final long asyncBufferTimeout =
                 environment.getExecutionConfig().getAsyncStateBufferTimeout();
-        // TODO: initial state executor and set state executor for aec
+
+        AsyncKeyedStateBackend asyncKeyedStateBackend =
+                Preconditions.checkNotNull(
+                        stateHandler.getAsyncKeyedStateBackend(),
+                        "Current State Backend doesn't support async access");
         this.asyncExecutionController =
                 new AsyncExecutionController(
                         mailboxExecutor,
-                        null,
+                        asyncKeyedStateBackend.createStateExecutor(),
                         maxParallelism,
                         asyncBufferSize,
                         asyncBufferTimeout,
                         inFlightRecordsLimit);
+        asyncKeyedStateBackend.setup(asyncExecutionController);
```

Comment:
the dependency looks like not very clear, can we make it like this?
```
new AsyncExecutionController(
                        mailboxExecutor,
                        asyncKeyedStateBackend,
                        maxParallelism,
                        asyncBufferSize,
                        asyncBufferTimeout,
                        inFlightRecordsLimit);
```
The AEC is responsible for creating the `stateExecutor` and set it up inside the construction method.

Reply:
Thanks for the suggestion.
I think it's reasonable to maintain `setup` in AEC.